### ISSUE TITLE
sample files 

### ISSUE THREAD

**Issue by athenas-lab** [czq142857/BSP-NET-pytorch#6]
Hi, 

Thank you for the code. I trained the AE model using train_ae.sh and using the prepared Shapenet dataset  that you have provided. I did not make any changes to the code, except pointing to the correct data_dir in main.py. After training with size 16 and 32, I looked at the sample files generated during training and testing. The generated sample .ply files  do not have any vertices, but all of them  just have the header as shown below. Is this the expected output? If not, could you please tell me how I can fix the problem and get the correct sample output with vertex info? The training logs are also shown below. 

Also, can the 4 progressive training steps (size 16 and 32 with 8M iterations and size 64 with 16M iterations) be run in parallel or do they have to be trained sequentially as shown in train_ae.sh?

Thank you.

---------sample ply file generated during training and testing----------
ply
format ascii 1.0
element vertex 0
property float x
property float y
property float z
element face 0
property list uchar int vertex_index
end_header
 
The following is a snippet of the training logs:
----------net summary----------
training samples    35019
-------------------------------

32 Epoch: [ 0/228] time: 171.8615, loss_sp: 0.252307, loss_total: 0.259578
32 Epoch: [ 1/228] time: 344.7302, loss_sp: 0.252319, loss_total: 0.259573
32 Epoch: [ 2/228] time: 517.7831, loss_sp: 0.252298, loss_total: 0.259548
32 Epoch: [ 3/228] time: 690.9526, loss_sp: 0.252300, loss_total: 0.259559
32 Epoch: [ 4/228] time: 864.0516, loss_sp: 0.252307, loss_total: 0.259560

**Comment by czq142857** [comment]
The loss is too large, therefore it seems the model has collapsed. Could you try training again from 16 and see if you can reproduce the problem? Does the loss at size 16 decrease?

The 4 progressive training steps cannot run in parallel. They have to be trained sequentially.

**Comment by athenas-lab** [comment]
Loss at size 16 decreases (somewhat rapidly). Here's a snippet of the logs at size 16:

16 Epoch: [ 0/228] time: 171.6810, loss_sp: 0.020969, loss_total: 959.547964
16 Epoch: [ 1/228] time: 343.9625, loss_sp: 0.011673, loss_total: 200.003053
16 Epoch: [ 2/228] time: 516.7791, loss_sp: 0.009346, loss_total: 162.652769
16 Epoch: [ 0/228] time: 171.6810, loss_sp: 0.020969, loss_total: 959.547964
16 Epoch: [ 1/228] time: 343.9625, loss_sp: 0.011673, loss_total: 200.003053
16 Epoch: [ 2/228] time: 516.7791, loss_sp: 0.009346, loss_total: 162.652769
16 Epoch: [ 3/228] time: 689.5728, loss_sp: 0.008031, loss_total: 125.303600
16 Epoch: [ 4/228] time: 862.3106, loss_sp: 0.007148, loss_total: 87.947247
16 Epoch: [ 5/228] time: 1034.9979, loss_sp: 0.006525, loss_total: 50.590102
16 Epoch: [ 6/228] time: 1207.6496, loss_sp: 0.006058, loss_total: 13.976282
16 Epoch: [ 7/228] time: 1380.2313, loss_sp: 0.005661, loss_total: 0.031156
16 Epoch: [ 8/228] time: 1552.7917, loss_sp: 0.005344, loss_total: 0.012295

The training currently runs only on a single GPU. Would there be any problems if we train using more gpus (using pytorch's DataParallel(model, gpulist)).
thanks

**Comment by czq142857** [comment]
Yes. The loss should look like this. You should check the output samples and see if they represent some coarse shapes.

I do not see a problem running on multiple GPUs, but I have not tried it yet, so I am not sure.

**Comment by athenas-lab** [comment]
I reran the training with size 16 and this time the .ply sample files have vertices. I was able to view the coarse shapes. Not sure why it did not work the first time. Is the uncertainty caused by some initialization values?

I am currently rerunning size 32, and here are some initial loss values, which appear to be around 1/10th of the values reported above for the first attempt
32 Epoch: [ 0/228] time: 169.3450, loss_sp: 0.024555, loss_total: 0.035554
32 Epoch: [ 1/228] time: 340.6203, loss_sp: 0.020585, loss_total: 0.030386
32 Epoch: [ 2/228] time: 512.6848, loss_sp: 0.018955, loss_total: 0.027908
32 Epoch: [ 3/228] time: 684.9327, loss_sp: 0.017964, loss_total: 0.026484
32 Epoch: [ 4/228] time: 857.2505, loss_sp: 0.017251, loss_total: 0.025372

Is it possible to provide  the expected initial loss values for size 32 and size 64 (phase 0 and 1) so that I can do a sanity check as the training progresses?

thanks

**Comment by czq142857** [comment]
The uncertainty is indeed caused by the weight initialization.

You do not need sanity checks for 32 and 64 -- if 16 is fine, then other resolutions will be fine.
Also, you only need to monitor the loss to see whether it is decreasing in order to determine whether there is a problem in training.

**Comment by athenas-lab** [comment]
Thank you. The training completed successfully.
